content/applications/sales/sales/invoicing/milestone.rst

```diff
@@ -2,32 +2,244 @@
+    workflow. However, a project *must* then be created directly from a sales order form with that
+    specific product. Once a project is created *then* milestones and tasks can be created and
+    configured.
+
+When the :guilabel:`Create on Order` default option of :guilabel:`Nothing` is clicked, a drop-down
+menu is revealed with the following options:
+
+- :guilabel:`Task`: Odoo creates a task related to this milestone product in the *Projects* app when
+  this specific product is ordered.
+- :guilabel:`Project & Task`: Odoo creates a project and task related to this milestone product in
+  the *Projects* app when this specific product is ordered.
+- :guilabel:`Project`: Odoo creates a project related to this milestone product in the *Projects*
+  app when this specific product is ordered.
+
+When :guilabel:`Task` is selected, a :guilabel:`Project` field appears. In this field, select to
+which existing project in the *Projects* app this created task should be connected.
+
+.. image:: milestone/task-option-project-field.png
+   :align: center
+   :alt: The Project field appears when the Task option is selected in Create on Order field.
+
+When :guilabel:`Project & Task` or :guilabel:`Project` is selected, two new fields appear:
+:guilabel:`Project Template` and :guilabel:`Workspace Template`.
 
-You have to set the product type as *Service* under general
-information and select *Milestones* in the sales tab.
+.. image:: milestone/project-task-option-project-workspace-fields.png
+   :align: center
+   :alt: The Project template and workspace template fields that appear on milestone product.
 
-.. image:: milestone/milestone01.png
-    :align: center
+The :guilabel:`Project Template` field provides template options to use for the project that will be
+created when this specific product is ordered.
+
+The :guilabel:`Workspace Template` field (for the *Documents* app, not the *Projects* app) provides
+template options to use for the workspace that will be automatically generated for the project when
+this specific product is ordered.
+
+.. tip::
+   For organizational purposes, click the :guilabel:`Sales` tab on the product form, and enter a
+   custom 'Milestone' related descriptor in the :guilabel:`Sales Description` field. This
+   information appears in the :guilabel:`Description` column on the :guilabel:`Order Lines` tab of
+   the sales order.
+
+   Or, directly edit/modify the :guilabel:`Description` field on the :guilabel:`Order Lines` tab of
+   the sales order.
+
+   This is *not* a requirement.
 
 Invoice milestones
 ==================
 
-From the sales order, you can manually edit the quantity delivered as
-you complete a milestone.
+.. note::
+   The following flow features a trio of milestone products that have :guilabel:`Service` set as
+   their :guilabel:`Product Type`, and :guilabel:`Task` set on their :guilabel:`Create on Order`
+   field.
```

Comment:
while creating the product to test your flow, I wasn't too sure which boxes to tick. I'm not sure if it's necessary to specify this information here-- I just wanted to point out everything I observed while following your doc (:
![image](https://github.com/odoo/documentation/assets/23611711/3766e5c2-3b44-4526-ac16-f0a7f1e3f64d)

Reply:
As always, I sincerely appreciate your attention to detail. If memory serves, I don't believe it's required to tick any of these boxes, specifically. So, I think for the sake of simplicity, I'll just omit such a sentence to avoid any confusion. But, once again, thanks for pointing this out. Also, if you *do* think it's important to mention this, just let me know, and I'll add it in right away. Thanks! 👍